src/ocl/clhelper.cpp

```diff
@@ -189,6 +192,15 @@ ClProgramPtr LoadProgram(cl_context ctx,
         bin_file_to_str(hsaco_file, buf);
         return LoadBinaryProgram(ctx, device, buf);
     }
+#if MIOPEN_USE_MLIR
+    else if(miopen::EndsWith(program_name, ".mlir"))
+    {
+        std::vector<char> buffer;
+        MiirGenBin(params, buffer);
+        std::string buf(buffer.begin(), buffer.end());
+        return LoadBinaryProgram(ctx, device, buf);
```

Comment:
Please create `LoadBinaryProgram()` that accepts `const std::vector<char>&` and eliminate redundant copying. Eventually we must switch to using `std::vector<char>` for all blobs, but this is not so yet.